07week/yahoo-fetch-api/yahoo.js

```diff
@@ -0,0 +1,99 @@
+function getStockData() {
+
+    let inputbox = document.getElementById("tickr");
+    let tickr = inputbox.value;
+
+    fetch("https://apidojo-yahoo-finance-v1.p.rapidapi.com/stock/get-detail?region=US&lang=en&symbol="+tickr, {
+        "method": "GET",
+        "headers": {
+            "x-rapidapi-host": "apidojo-yahoo-finance-v1.p.rapidapi.com",
+            "x-rapidapi-key": "aa58a61455msh9a74d8a8fc41f2fp1c9e39jsn6213f7c23f2a"
+        }
+    })
+    .then(response => {
+        console.log(response);
+        return response.json();
+    })
+    .then(myjson => {
+        console.log(myjson)
+        defineVariables(myjson.Object)
+    })
+      
+    .catch(err => {
+        console.log(err);
+    });
+  
+}
+
+// Define the Important Company Return Metrics
+// Defining the values of the variables collected to be updated
+    // This is the variable definitions to get the value that 
+    // will be used to update the spad with matching element
+    // it looks like the term is JSON handeling
+    // which of these is correct, above format, or below...?
+    // I moved these variable defining statements UP
+    // to inside the console log of the my json promise
+
+function defineVariables() {
+    let companyName = Object.price.shortName;
```

Comment:
When you defined variables in a function or method using 'let' or 'var' those variables are only available inside that function unless you pass them as input to another function